src/AspNet.Security.OAuth.EVEOnline/EVEOnlineAuthenticationHandler.cs

```diff
@@ -20,44 +22,90 @@ namespace AspNet.Security.OAuth.EVEOnline
-                                /* Status: */ response.StatusCode,
-                                /* Headers: */ response.Headers.ToString(),
-                                /* Body: */ await response.Content.ReadAsStringAsync());
-
-                throw new HttpRequestException("An error occurred while retrieving the user profile.");
+                identity.AddClaim(claim);
             }
 
-            using var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync());
-
             var principal = new ClaimsPrincipal(identity);
-            var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, payload.RootElement);
+            var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, tokens.Response.RootElement);
             context.RunClaimActions();
 
             await Options.Events.CreatingTicket(context);
             return new AuthenticationTicket(context.Principal, context.Properties, Scheme.Name);
         }
+
+        /// <summary>
+        /// Extracts the claims from the token received from the token endpoint.
+        /// </summary>
+        /// <param name="token">The token to extract the claims from.</param>
+        /// <returns>
+        /// An <see cref="IEnumerable{Claim}"/> containing the claims extracted from the token.
+        /// </returns>
+        protected virtual IEnumerable<Claim> ExtractClaimsFromToken([NotNull] string token)
+        {
+            try
+            {
+                var securityToken = _tokenHandler.ReadJwtToken(token);
+
+                var claims = new List<Claim>(securityToken.Claims)
+                {
+                    new Claim(ClaimTypes.NameIdentifier, securityToken.Subject.Replace("CHARACTER:EVE:", string.Empty, StringComparison.OrdinalIgnoreCase), ClaimValueTypes.String, ClaimsIssuer),
+                    new Claim(ClaimTypes.GivenName, securityToken.Claims.First(x => x.Type.Equals("name", StringComparison.OrdinalIgnoreCase)).Value, ClaimValueTypes.String, ClaimsIssuer),
+                    new Claim(ClaimTypes.Name, securityToken.Claims.First(x => x.Type.Equals("name", StringComparison.OrdinalIgnoreCase)).Value, ClaimValueTypes.String, ClaimsIssuer),
+                    new Claim(ClaimTypes.Expiration, UnixTimeStampToDateTime(securityToken.Claims.First(x => x.Type.Equals("exp", StringComparison.OrdinalIgnoreCase)).Value), ClaimValueTypes.DateTime, ClaimsIssuer),
+                };
+
+                var scopes = claims.Where(x => x.Type.Equals("scp", StringComparison.OrdinalIgnoreCase)).ToList();
+
+                if (scopes.Any())
+                {
+                    claims.RemoveAll(x => scopes.Contains(x));
+
+                    claims = claims.Append(new Claim(EVEOnlineAuthenticationConstants.Claims.Scopes, string.Join(" ", scopes.Select(x => x.Value)), ClaimValueTypes.String, ClaimsIssuer)).ToList();
```

Comment:
You already have a list here so you can just add the one value to it:

```suggestion
                    claims.Add(new Claim(EVEOnlineAuthenticationConstants.Claims.Scopes, string.Join(" ", scopes.Select(x => x.Value)), ClaimValueTypes.String, ClaimsIssuer));
```